### PR TITLE
Stage 8 of clang compiler warning patches.

### PR DESCRIPTION
**Pull request by ethoms**
Stage 8: *CAUTION - needs testing*

Fixes ambiguous boolean logic caused by lack of brackets between AND and OR expressions. Since it's so ambiguous, I had to guess what was the correct logic. Without knowing the user-agent strings involved, this was my best estimation. Note that Ludovic reckons it looks OK in an email conversation. Ideally it needs testing.

### CODE REVIEW

File: SoObjects/SOGo/WORequest+SOGo.m [hunk 1]
```diff
@@ -180,14 +180,24 @@ - (BOOL) isMacOSXAddressBookApp
...
+            [[cc userAgent] rangeOfString: @"CFNetwork"].location != NSNotFound
+            &&
+            (
+              [[cc userAgent] rangeOfString: @"Mac OS X"].location != NSNotFound
+              ||
+              [[cc userAgent] rangeOfString: @"Mac_OS_X"].location != NSNotFound
+            )
+          )
+        )
+        && [[cc userAgent] rangeOfString: @"AddressBook"].location != NSNotFound
```
Comment (cgx) [inline SoObjects/SOGo/WORequest+SOGo.m]: The address book app on OSX 10.6 uses a localized user-agent string so we can't expect to locate **AddressBook**. Even the English localization will use **Address%20Book** as shown in the comments above the method.

**Comment by ethoms** [comment]
OK, noted. I'll look at those comments. I'd still like to clear the compiler warnings. Even if that just means adding brackets. Does the current code work properly? If so, I'd need to work out how the  compiler interprets the logic and add brackets accordingly. I remember in maths the '+' is compounded with priority. E.g. "1 + 2 x 3" is corrected to  "(1 + 2) x 3", as opposed to "1 + (2 x 3)". So I need to find the same rules for boolean logic, it's been a while since I learnt this at university.

**Comment by ethoms** [comment]
I only just noticed the comments on the user agent strings. Silly me, appologies.

Good, so now I can write the logic up better. Those kibd of comments are worth gold!

**Comment by cgx** [comment]
The original condition is working properly. The boolean computation was:

`A && B || ( A || C ) && D`

which is really:

`( A && B ) || ( ( A || C ) && D )`

and could be simplified to:

`( A && B ) || ( A && D ) || ( C && D )`

**Comment by extrafu** [comment]
The simplified form is the way to go!

**Comment by ethoms** [comment]
OK, I have something a bit different to try out. It should be easier to read code, less lines of code and easier to extend in future. But I'm not sure if the logic is 100% correct. I'll edit this PR, and you tell me if it looks OK.

**Comment by cgx** [comment]
Please just use my simplified form. Your latest commit still assume that *Address%20Book* is not localized for OS X 10.6 which is wrong.

Thanks for your hard work!

**Comment by ethoms** [comment]
Sorry, I forgot about the localized string. I've corrected as you suggested:

    ( A && B ) || ( A && D ) || ( C && D )

**Pull request merged**